### DISCUSSION TITLE
Call main script as a Python entrypoint

### DISCUSSION THREAD

**Discussion by HealthyPear** [zauberzeug/nicegui#3769] [Q&A]
### Question

I am trying to test a gui for a package I am developing.

I want the gui to be a "plus" for the moment, while the application can still be run as a script.

I though to make a simple example like this

`src/my_package/gui/main.py`:

```python
from nicegui import ui

def main():
    ui.label("Hello NiceGUI!")
    ui.run()

if __name__ == "__main__":
    main()
```

`pyproject.toml`:

```toml
[project.scripts]
my-package = "my_package.scripts.main:main"
my-package-gui = "my_package.gui.main:run"
```

but I get,

```
You must call ui.run() to start the server.
If ui.run() is behind a main guard
   if __name__ == "__main__":
remove the guard or replace it with
   if __name__ in {"__main__", "__mp_main__"}:
to allow for multiprocessing.
```

So I go for the second option,

```python
from nicegui import ui

def main():
    ui.label("Hello NiceGUI!")
    ui.run()

if __name__ in {"__main__", "__mp_main__"}:
    main()
```

and I still get the same error.

Am I missing something?

**Comment by HealthyPear** [comment]
Ah wait I have to call `ui.run()` with `reload=False` to make it work.

Any way one can get both things?
Even though the docs say 
`reload: | automatically reload the UI on file changes (default: True)`
are these files the source code files?
is this option only meant for testing/debugging the web app?


https://github.com/zauberzeug/nicegui/issues/181#issuecomment-1328965228

**Reply by falkoschindler** [reply]
Yes, I think the trick is to set `reload=False`. This flag is `True` by default to automatically reload the GUI when changing the source code, allowing to iterate quickly when building a NiceGUI app. For Python entrypoints this is probably not necessary.

**Comment by rodja** [comment]
NiceGUI starts creating objects and stuff when importing. This works for me:

```py
import sys

def main():
    print('console main')

def gui():
    from nicegui import ui
    ui.label("GUI main!")
    ui.run()

if __name__ in {"__main__", "__mp_main__"}:
    if sys.argv[1:] == ['gui']:
        gui()
    else:
        main()
```

If started without parameter it simply prints "console main". But if started with "gui" as parameter, the website opens.

**Comment by yihuang** [comment]
```python
def main(reload=False):
    ui.run(reload=reload)

if __name__ in {"__main__", "__mp_main__"}:
    main(True)
```

```toml
[project.scripts]
nicechat = "pkg.app:main"
```

So in development I run `python -mpkg.app`, it'll run in reload mode.
And script entrypoint will also work, althrough it has `if __name__ == "__main__":`.

**Comment by falkoschindler** [comment]
The problem might have been fixed with PR #6068 which just have been released in 3.13.
